src/uct/cuda/cuda_copy/cuda_copy_md.c

```diff
@@ -221,10 +221,11 @@ uct_cuda_copy_mem_alloc_fabric(uct_cuda_copy_md_t *md,
         return status;
     }
 
-    prop.type                 = CU_MEM_ALLOCATION_TYPE_PINNED;
-    prop.requestedHandleTypes = CU_MEM_HANDLE_TYPE_FABRIC;
-    prop.location.type        = CU_MEM_LOCATION_TYPE_DEVICE;
-    prop.location.id          = cu_device;
+    prop.type                            = CU_MEM_ALLOCATION_TYPE_PINNED;
+    prop.requestedHandleTypes            = CU_MEM_HANDLE_TYPE_FABRIC;
+    prop.location.type                   = CU_MEM_LOCATION_TYPE_DEVICE;
+    prop.location.id                     = cu_device;
+    prop.allocFlags.gpuDirectRDMACapable = 1;
```

Comment:
Will the allocation fail on a system with no IB?

Reply:
@SeyedMir 
No. `gpuDirectRDMACapable` is a request. CUDA isn't obliged to honor it and report errors if rdma enablement fails. Instead user is expected to check attribute `CU_POINTER_ATTRIBUTE_IS_GPU_DIRECT_RDMA_CAPABLE` once allocation succeeds to see if RDMA capability is enabled. 

But thanks for raising this as we do need to check that allocation has handle fabric allowed with it and we should return a failure if that isn't possible. I've made relevant changes now.

Reply:
What about systems without RDMA (i.e TCP+shared mem only), maybe passing such flag could lead to failure?

Reply:
Maybe we should not fail if the allocation is not RDMA capable, and let it fail later if actually used with RDMA